### PR TITLE
Prepare version 5.0.0

### PR DESCRIPTION
**Pull request by glasserc**
This release:

- Massive refactor of most methods from manually-handled promises to async/await (#538).
- Updates the documentation with a "NOP" schema example for use when you don't really need an `idSchema` (#533).
- Optimize `collection#list` by doing filtering before putting documents in a list rather than after (#543).
- Upgrades tests to Kinto 4.3 (#573).
- Fix for a puzzling bug involving IndexedDB use on Safari (#549).
- Fixes yet another bug in conflict resolution during `CLIENT_WINS` when using transformers (#570).
- Add a `FirefoxAdapter#resetSyncStatus` that resets sync status across all collections, rather than the one you happen to have gotten your hands on (#571). This is not possible as yet in IndexedDB, so it's only in `FirefoxAdapter` for now.

This might be the last release before we take `FirefoxStorage` out of this repo and move it to Gecko.

r? @leplatrem @Natim

### CODE REVIEW

**Comment by Natim** [comment]
> This might be the last release before we take FirefoxStorage out of this repo and move it to Gecko.

Final last words :D

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "kinto",
-  "version": "4.0.4",
```
Comment (Natim) [inline package.json]: Did we really released a 4.0.4 version of Kinto.js? I didn't find it...

Comment (n1k0) [inline package.json]: yup https://github.com/Kinto/kinto.js/releases/tag/v4.0.4

**Comment by n1k0** [comment]
fwiw, lgtm

File: package.json (hunk 1 above)
Comment (Natim) [inline package.json]: /me was joking of course 👍

Comment (n1k0) [inline package.json]: /me grabs moar coffee

**Pull request merged**